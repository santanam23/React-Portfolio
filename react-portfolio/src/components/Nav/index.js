import React from "react";

function Nav() {
  return (
    <header>
      <div className="navbar">
        <h2>
          <a data-testid="link" href="/">
            {/* <span role="img" aria-label="camera"></span>  */}
            Miriams Portfolio
          </a>
        </h2>
        <nav>
          <ul className="flex-row">
            <li className="mx-2">
              <a data-testid="home" href="#home">
                Home
              </a>
            </li>
            <li className="mx-2">
              <a data-testid="about" href="#about">
                About me
              </a>
            </li>
            <li className="mx-2">
              <a data-testid="contact" href="#contact">
                Contact
              </a>
            </li>
            <li className="mx-2">
              <a data-testid="project" href="#project">
                Project
              </a>
            </li>
            <li className="mx-2">
              <a data-testid="resume" href="#resume">
                Resume
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
}
export default Nav;
