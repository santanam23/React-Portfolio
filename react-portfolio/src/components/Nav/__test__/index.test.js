import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);
//use state allows us to manage in  use effect hook
describe('Nav component', () => {
    //baseline test
    it('renders', () => {
        render(<Nav/>);
    });
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });
    describe('links are visible', () => {
        it('inserts text into the links', () => {
          const { getByTestId } = render(<Nav />);
          expect(getByTestId('link')).toHaveTextContent('Oh Snap!');
          expect(getByTestId('about')).toHaveTextContent('About me');
        });
      });
      
    // emoji visibility
    // describe('emoji is visible', () => {
    //     it('inserts emoji into the h2', () => {
    //     // Arrange
    //     const { getByLabelText} = render(<Nav />);
    //     // Assert 
    //     expect(getByLabelText('camera')).toHaveTextContent('emoji');
    //     })
    // })
})