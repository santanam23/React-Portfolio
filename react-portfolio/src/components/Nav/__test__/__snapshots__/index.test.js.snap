// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        Miriams Portfolio
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="contact"
            href="#contact"
          >
            Contact
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="project"
            href="#project"
          >
            Project
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="resume"
            href="#resume"
          >
            Resume
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
